File: Programmers/1/140108. 문자열 나누기/문자열 나누기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string s) {
    int answer = 0;
    int same = 0;
    int diff = 0;
    char x = s[0];

    for (int i = 0; i < s.size(); i++)
    {
        if (s[i] == x)
            same++;
        else
            diff++;

        if (same == diff)
        {
            answer++;
            same = 0;
            diff = 0;

            if (i + 1 < s.size())
                x = s[i + 1];
        }
    }

    if (same != diff)
        answer++;
    
    return answer;
}

File: Programmers/1/140108. 문자열 나누기/문자열 나누기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int answer = 0;
        int same = 0;
        int diff = 0;
        char x = s.charAt(0);
        
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == x) {
                same++;
            } else {
                diff++;
            }
            
            if (same == diff) {
                answer++;
                same = 0;
                diff = 0;
                
                if (i + 1 < s.length()) {
                    x = s.charAt(i + 1);
                }
            }
        }
        
        if (same != diff) {
            answer++;
        }
        
        return answer;
    }
}